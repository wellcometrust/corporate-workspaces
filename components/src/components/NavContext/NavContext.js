// @flow
import React, { createContext, useState } from 'react';
import type { Node } from 'react';

const defaultState = {
  isNavActive: false,
  openNav: () => {}
};

type ContextProps = {|
  isNavActive: boolean,
  openNav: (bool: boolean) => void
|};

type ProviderProps = {|
  children: Node
|};

const NavContext = createContext<ContextProps>(defaultState);

export const NavContextProvider = ({ children }: ProviderProps) => {
  const [state, setState] = useState({
    isNavActive: false,
    openNav: bool =>
      setState(prevState => ({ ...prevState, isNavActive: bool }))
  });

  return <NavContext.Provider value={state}>{children}</NavContext.Provider>;
};

export default NavContext;
