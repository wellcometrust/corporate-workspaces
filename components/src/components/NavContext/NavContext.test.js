import React from 'react';
import { mount } from 'enzyme';

import '../../test/mocks/match-media.mock';
import NavContext, { NavContextProvider } from './NavContext';

describe('<NavContextProvider />', () => {
  let output;

  beforeEach(() => {
    output = mount(
      <NavContextProvider>
        <NavContext.Consumer>
          {({ isNavActive }) => (
            <div className="test-isNavActive">{isNavActive.toString()}</div>
          )}
        </NavContext.Consumer>
      </NavContextProvider>
    );
  });

  afterEach(() => {
    output.unmount();
  });

  it('renders the component', () => {
    expect(output).toMatchSnapshot();
  });

  it('has `isNavActive` property set to false', () => {
    expect(output.find('.test-isNavActive').text()).toBe('false');
  });
});
