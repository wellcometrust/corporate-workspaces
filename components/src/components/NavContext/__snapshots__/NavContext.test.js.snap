// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavContextProvider /> renders the component 1`] = `ReactWrapper {}`;
